cache-tree.c

```diff
@@ -110,7 +110,7 @@ static int do_invalidate_path(struct cache_tree *it, const char *path)
 	int namelen;
 	struct cache_tree_sub *down;
 
-#if DEBUG
+#if 0 && DEBUG
```

Comment:
I am pretty certain that the `DEBUG` constant does not have the same meaning in `gcc` as it has in MSVC. As such, I believe that the original author meant for this constant to be `#define`d by a developer within the file (see 749864627c2d3c33bbc56d7ba0b5542af698cc40 for the original contribution that seems to concur with my assessment).

Therefore, I think the best course of action would be to rename all instances of `DEBUG` to `DEBUG_CACHE_TREE` in this file.